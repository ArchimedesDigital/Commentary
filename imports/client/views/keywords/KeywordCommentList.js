const KeywordCommentList = (props) => {

	console.log('props', props)

	if (props.keywordComments) {
		return (
			<div>
				{props.keywordComments.map((comment, i) => (
					<article
						className="comment commentary-comment paper-shadow "
					>
						TEST
					</article>
				))}
			</div>);
	}
	return null;
};
KeywordCommentList.propTypes = {
	keywordComments: React.PropTypes.array,
};

export default KeywordCommentList;
