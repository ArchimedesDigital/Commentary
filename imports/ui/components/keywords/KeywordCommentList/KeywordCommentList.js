import React from 'react';

// api
import Commenters from '/imports/api/collections/commenters';

// components
import AvatarIcon from '/imports/ui/components/avatar/AvatarIcon';

// lib
import createRevisionMarkup from '/imports/lib/createRevisionMarkup';

const KeywordCommentList = (props) => {

	if (props.keywordComments) {
		return (
			<div className="comment-outer has-discussion">
				{props.keywordComments.map((comment, i) => {
					const selectedRevision = comment.revisions.pop();
					const updated = selectedRevision.updated;
					const format = 'D MMMM YYYY';

					return (
						<article
							key={comment._id}
							className="comment commentary-comment paper-shadow "
						>

							<div className="comment-upper">
								<div className="comment-upper-left">
									<h2 className="comment-title" style={{margin: 0, padding: 0}}><a href={`/commentary?_id=${comment._id}`}>{selectedRevision.title}</a></h2>
								</div>

								<div className="comment-upper-right">
									{comment.commenters.map((commenter, _i) => {
										const commenterRecord = Commenters.findOne({
											slug: commenter.slug,
										});

										if (!commenterRecord) {
											return null;
										}

										return (
											<div
												key={_i}
												className="comment-author"
											>
												<div className="comment-author-text">
													<a href={`/commenters/${commenterRecord.slug}`}>
														<span className="comment-author-name" style={{margin: 0, textAlign: 'right'}}>{commenterRecord.name}</span>
													</a>
													<span className="comment-date" style={{margin: 0, textAlign: 'right'}}>
														{moment(updated).format(format)}
													</span>
												</div>
												<div className="comment-author-image-wrap paper-shadow">
													<a href={`/commenters/${commenterRecord.slug}`}>
														<AvatarIcon
															avatar={
																(commenterRecord && 'avatar' in commenterRecord) ?
																commenterRecord.avatar.src
																: null
															}
														/>
													</a>
												</div>
											</div>
										);
									})}
								</div>
							</div>

							<div className="comment-lower">
								<div
									className="comment-body"
									dangerouslySetInnerHTML={createRevisionMarkup(selectedRevision.text)}
								/>
							</div>
						</article>
					);
				})}
			</div>);
	}
	return null;
};
KeywordCommentList.propTypes = {
	keywordComments: React.PropTypes.array,
};

export default KeywordCommentList;
