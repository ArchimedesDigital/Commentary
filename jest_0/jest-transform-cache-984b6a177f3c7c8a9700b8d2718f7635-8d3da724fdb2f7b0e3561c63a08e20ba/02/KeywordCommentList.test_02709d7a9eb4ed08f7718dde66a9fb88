/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_KeywordCommentList = require('./KeywordCommentList'); /* istanbul ignore next */var _KeywordCommentList2 = _interopRequireDefault(_KeywordCommentList);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('KeywordCommentList', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_KeywordCommentList2.default, /* istanbul ignore next */null)).
		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: