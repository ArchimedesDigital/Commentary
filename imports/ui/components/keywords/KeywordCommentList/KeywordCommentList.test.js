import React from 'react';
import renderer from 'react-test-renderer';

// component:
import KeywordCommentList from './KeywordCommentList';

describe('KeywordCommentList', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<KeywordCommentList />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
