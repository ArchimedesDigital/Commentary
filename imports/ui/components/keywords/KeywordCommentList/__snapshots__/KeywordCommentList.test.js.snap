// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordCommentList renders correctly 1`] = `
<div
  className="comment-outer has-discussion"
>
  <div
    className="no-results"
  >
    This tag has no comments associated with it yet.
  </div>
</div>
`;
